pkg/ibm/ibmz.go

```diff
@@ -129,6 +129,35 @@ func (r IBMZDynamicConfig) CountInstances(kubeClient client.Client, log *logr.Lo
 	return count, nil
 }
 
+func (r IBMZDynamicConfig) ListInstances(kubeClient client.Client, log *logr.Logger, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
+	vpcService, err := r.authenticate(kubeClient, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	vpc, err := r.lookupVpc(vpcService)
+	if err != nil {
+		return nil, err
+	}
+	instances, _, err := vpcService.ListInstances(&vpcv1.ListInstancesOptions{ResourceGroupID: vpc.ResourceGroup.ID, VPCName: &r.Vpc})
+	if err != nil {
+		return nil, err
+	}
+	ret := []cloud.CloudVMInstance{}
+	for _, instance := range instances.Instances {
+		if strings.HasPrefix(*instance.Name, instanceTag) {
```

Comment:
Is this instanceTag defined in the multi-platform deployment on each cluster itself so that we can identify the instances in a pool uniquely between all platforms? 

I think that it is possible to _label_ servers in IBM Cloud. Is leveraging these much more complex than just identifying a pattern on the instance name?

Reply:
For IBM I just use it as part of the instance name.